src/local/butler/deploy.py

```diff
@@ -91,7 +91,7 @@ def _deploy_app_prod(project,
                      package_zip_paths,
                      deploy_appengine=True):
   """Deploy app in production."""
-  if deploy_appengine:
+  if deploy_appengine and yaml_paths:
```

Comment:
nit: "deploy_appengine and yaml_paths" can be moved to a local and use in three places

Reply:
Just set `deploy_appengine = False` if there are no yaml paths.